app/screens/edit_profile/edit_profile.test.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {act} from '@testing-library/react-hooks';
+import {fireEvent} from '@testing-library/react-native';
+import React from 'react';
+
+import AvailableScreens from '@constants/screens';
+import {renderWithIntlAndTheme} from '@test/intl-test-helper';
+
+import EditProfile from './edit_profile';
+
+import type {UserModel} from '@database/models/server';
+
+// Mock CompassIcon
+jest.mock('@components/compass_icon', () => {
+    function CompassIcon() {
+        return null;
+    }
+    CompassIcon.getImageSourceSync = jest.fn().mockReturnValue({});
+    return {
+        __esModule: true,
+        default: CompassIcon,
+    };
+});
+
+// Mock server context
+jest.mock('@context/server', () => ({
+    useServerUrl: jest.fn().mockReturnValue('http://localhost:8065'),
+}));
+
+// Mock remote user actions
+jest.mock('@actions/remote/user', () => ({
+    fetchCustomAttributes: jest.fn().mockResolvedValue({
+        attributes: {
+            attr1: {
+                id: 'attr1',
+                name: 'Custom Attribute 1',
+                value: 'original value',
+                sort_order: 1,
+            },
+        },
+        error: undefined,
+    }),
+    updateCustomAttributes: jest.fn().mockResolvedValue({}),
+    updateMe: jest.fn().mockResolvedValue({}),
+    uploadUserProfileImage: jest.fn().mockResolvedValue({}),
+    setDefaultProfileImage: jest.fn().mockResolvedValue({}),
+    buildProfileImageUrlFromUser: jest.fn().mockReturnValue('http://example.com/profile.jpg'),
+}));
+
+describe('EditProfile', () => {
+    const mockCurrentUser = {
+        id: 'user1',
+        email: 'test@example.com',
+        firstName: 'John',
+        lastName: 'Doe',
+        nickname: 'Johnny',
+        position: 'Developer',
+        username: 'johndoe',
+    } as UserModel;
```

Comment:
Soon we will be able to create these mocks with the TestHelper :P https://github.com/mattermost/mattermost-mobile/pull/8715